bpf/lib/fib.h

```diff
@@ -124,36 +121,39 @@ fib_do_redirect(struct __ctx_buff *ctx, const bool needs_l2_check,
 		 * the latter is used if we don't and have to use the eBPF
 		 * neighbor map.
 		 */
-		if (fib_params) {
-			nh_params.nh_family = fib_params->l.family;
-			__bpf_memcpy_builtin(&nh_params.ipv6_nh,
-					     &fib_params->l.ipv6_dst,
-					     sizeof(nh_params.ipv6_nh));
-			nh = &nh_params;
-
-			if (!neigh_resolver_available()) {
-				/* The neigh_record_ip{4,6} locations are mainly from
-				 * inbound client traffic on the load-balancer where we
-				 * know that replies need to go back to them.
-				 */
-				dmac = fib_params->l.family == AF_INET ?
-				neigh_lookup_ip4(&fib_params->l.ipv4_dst) :
-				neigh_lookup_ip6((void *)&fib_params->l.ipv6_dst);
-			}
-		}
+
 
 		/* If we are able to resolve neighbors on demand, always
 		 * prefer that over the BPF neighbor map since the latter
 		 * might be less accurate in some asymmetric corner cases.
 		 */
 		if (neigh_resolver_available()) {
```

Comment:
I'm a bit confused about this if statement. You no longer value the 'use_neigh_map' boolean. If we want to move toward being "more explicit" oh what the function is doing based on boolean arguments, doesn't this throw an override on the expected behavior when the function is called?

Reply:
It's intended behaviour - when fib_redirect() is available, we always want to prefer it over the neigh map.

I'll add proper docs for all the parameters. We can also call it `allow_neigh_map_fallback` or such.

Reply:
Yeah maybe that variable name update would suggest whats happening better.